src/RoomsSDKAdapter.js

```
@@ -134,4 +144,119 @@ export default class RoomsSDKAdapter extends RoomsAdapter {
+
+    if (activityIds[earliestActivityDate]) {
+      return activityIds[earliestActivityDate] === this.limit;
+    }
+
+    return true;
+  }
+
+  /**
+   * Fetches past activities and returns array of (id, published) objects.
+   *
+   * @param {string} roomId The id of the room
+   * @param {number} limit The max number of activities to return
+   * @returns {external:Observable.<Activity>} Observable stream that emits activity data
+   */
+  async fetchPastActivities(roomId) {
+    const roomActivity = this.roomActivities.get(roomId);
+    let {earliestActivityDate} = roomActivity;
+    const roomObservable = this.roomObservables[roomId];
+
+    if (!this.hasMoreActivities(roomId)) {
+      return roomObservable.complete(roomActivity);
+    }
+
+    try {
+      const data = await this.fetchActivities({
+        roomId,
+        earliestActivityDate,
+      });
+
+      if (!data) {
+        return roomObservable.complete(roomActivity);
+      }
+
+      const activityIds = data.map((activity) => {
+        const {id} = activity;
+
+        roomActivity.activities.set(id, activity);
+
+        return id;
+      });
+
+      const [{published}] = data;
+
+      earliestActivityDate = published;
+      roomActivity.earliestActivityDate = published;
+      roomActivity.activityIds[earliestActivityDate] = activityIds.length;
+
+      this.roomActivities.set(roomId, roomActivity);
+
+      return roomObservable.next(activityIds);
+    } catch (err) {
+      return roomObservable.error(err);
+    }
+  }
+
+  /**
+   * Handle subscribing to previous activities fetched.
+   *
+   * @param {string} roomId The id of the room
+   * @param {number} count The maximum number of activities to return
+   * @returns {external:Observable.<Room>} Observable stream that emits activity data
+   */
+  getPreviousActivities(roomId, count = 50) {
+    this.limit = count;
+    if (!roomId) {
+      throw new Error('Must provide roomId!');
```

This should throw an error through the observable. Please also make it clearer on the error message where this error is happening

---

`getPreviousActivities` should only return an observable, if there are errors, the observable should error